project2/task2_2/deterministicHopfield.py

```diff
@@ -22,9 +21,20 @@ def deterministic_behavior(Class):
     Class.asynchronous_choice = asynchronous_choice
 
 
-if __name__ == '__main__':
-    k = 3
-    x = kRooksHopfield(k)
+def compare(k, n, criteria):
+    deterministic = kRooksHopfield(k)
+    nonDeterministic = kRooksHopfield(k)
+    nonDeterministic.multiple_runs(n=n, convergence_params=[criteria])
     deterministic_behavior(kRooksHopfield)
-    x.run(convergence_params=[100])
-    print(x)
+    deterministic.run(convergence_params=[criteria])
+    return deterministic, nonDeterministic
+
+
+if __name__ == '__main__':
+    k = 4
+    criteria = 100
+    n = 10
+    x, y = compare(k, n, criteria)
+    print(x, y)
+    print("Deterministic state: {}, energy: {}\n".format(str(x.state), x.previous_energies[-1]))
```

Comment:
why not do these prints inside the compare function, as it's kind of part of its functionality.